editor/modes/visual-editor/block.js

```diff
@@ -283,6 +283,7 @@ class VisualEditorBlock extends Component {
 			], this.props.order + 1 );
 		}
 		this.removeOrDeselect( event );
+		this.maybeStartTyping( event );
```

Comment:
This is just to see how it feels—I like that arrow keys trigger "typing" mode now. But we'd want to exclude modifier keys, etc. cc @iseulde @mcsf

Reply:
Pushed 118f489ecace3f780a7bcf2ea8fe14d5ff2c24f6 to consolidate keydown handling, targeting arrow keys explicitly for triggering typing mode.

This reverts some previous refactoring that had split out `removeOrDeselect` to a separate function, but I think the switch is a bit easier to reason about consistently. Otherwise maybe some mapping of keys and conditions (`isTarget`) via something like [`cond`](https://lodash.com/docs/4.17.4#cond) but seems overkill and prone to a performance impact on this performance-critical code.

Reply:
Looks good to me.